Signal/src/AppDelegate.m

```diff
@@ -156,11 +154,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     // This block will be cleared in storageIsReady.
     [DeviceSleepManager.sharedInstance addBlockWithBlockObject:self];
 
-    [AppSetup setupEnvironmentWithCallMessageHandlerBlock:^{
-        return SignalApp.sharedApp.callMessageHandler;
-    }
-        notificationsProtocolBlock:^{
-            return SignalApp.sharedApp.notificationsManager;
+    [AppSetup
+        setupEnvironmentWithAppSpecificSingletonBlock:^{
+            [SignalApp.sharedApp createSingletons];
```

Comment:
There's now a `AppSpecificSingletonBlock` that lets SAE and main app create their singletons.